### PR TITLE
Try to use our thread pool instead of the bukkit async scheduler

### PR DESCRIPTION
**Pull request by sgdc3**
Needs refactor, @ljacqu what do you think?

### CODE REVIEW

**Comment by ljacqu** [comment]
hi @games647 @DNx5 — thoughts on this? I think you have deeper knowledge about scheduling than me.

For me it looks OK, but I personally wouldn't agree to merging this into master until two things are done:
- **performance measurement** of Bukkit thread pool vs. this one proving that this approach is better
- instead of introducing a new alt method, just replace the implementations of the existing method

Also, the ExecutorService could live inside BukkitService. At the latest if we decide to add a setting for Bukkit pool vs. our own it will make sense to have it in there.

**Comment by games647** [comment]
@sgdc3 Could you state why we really need this?

I think it has a negative effect, because we cannot access the inactive threads of bukkit. Instead we create a new pool which is separated with the one of bukkit.

**Comment by sgdc3** [comment]
Bukkit thread pool has no limits for current threads, this means 1 login = 2-3 threads

**Comment by sgdc3** [comment]
100 bot joins the server and register -> 300 threads are created

**Comment by sgdc3** [comment]
CachedThreadPool has a queue that help us to limit the max thread number based on the cpu core count.

File: src/main/java/fr/xephi/authme/AuthMe.java [hunk 1]
```diff
@@ -64,6 +66,9 @@
     private static String pluginVersion = "N/D";
     private static String pluginBuildNumber = "Unknown";
 
+    // The async thread pool
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```
Comment (games647) [inline src/main/java/fr/xephi/authme/AuthMe.java]: ATM, it's nearly the same as the bukkit one.

https://github.com/bergerkiller/CraftSource/blob/master/org.bukkit.craftbukkit/scheduler/CraftScheduler.java#L75

Bukkit just gives those threads a custom name.

**Comment by sgdc3** [comment]
:/ damn, i was sure the implementation wasn't a CachedThreadPool

**Comment by sgdc3** [comment]
This would be a solution:

```java
new ThreadPoolExecutor(CORE_COUNT, Integer.MAX_VALUE,
                                  60L, TimeUnit.SECONDS,
                                  new SynchronousQueue<Runnable>());
```

**Comment by sgdc3** [comment]
@games647

**Comment by sgdc3** [comment]
I'll open a new PR.

**Pull request closed without merging**